### PR TITLE
Added extrema to axes

### PR DESCRIPTION
**Pull request by lorem--ipsum**

### CODE REVIEW

File: src/utils/AxisOptions.ts [hunk 1]
```diff
@@ -26,12 +29,21 @@ module n3Charts.Utils {
...
-      this.type = js.type;
+      this.type = Options.getString(js.type, 'linear');
       this.key = js.key;
+
+      if (this.type === AxisOptions.TYPE.LINEAR) {
+        this.min = Options.getNumber(js.min, undefined);
+        this.max = Options.getNumber(js.max, undefined);
+      } else if (this.type === AxisOptions.TYPE.DATE) {
+        this.min = Options.getDate(js.min, undefined);
+        this.max = Options.getDate(js.max, undefined);
```
Comment (chaosmail) [inline src/utils/AxisOptions.ts]: yeah - this looks amazing :smile:

File: src/utils/Options.ts [hunk 2]
```diff
@@ -110,13 +110,23 @@ module n3Charts.Utils {
...
+    static getString(value: any, defaultValue?: string) {
+      var s = undefined;
+      if (value === undefined) {
+        s = String(defaultValue);
+      } else {
+        s = String(value);
+      }
+
       return s;
     }
```
Comment (chaosmail) [inline src/utils/Options.ts]: I don't know, but i would prefer like this:

```js
static getString(value: any, defaultValue?: string) {
   return value ? String(value) : String(defaultValue);
 }
```

File: test/spec/utils/Options.spec.ts [hunk 3]
```diff
@@ -184,6 +182,99 @@ describe('n3Charts.Utils.Options', () => {
...
+
+    it('should assume the axis is linear by default', () => {
+      var testing = axes[AxisOptions.SIDE.Y].type;
+      var expected = 'linear';
+
+      expect(testing).to.equal(expected);
+    });
+  });
+
+  describe('sanitizeAxes - numeric extrema', () => {
```
Comment (chaosmail) [inline test/spec/utils/Options.spec.ts]: I would love to make 2 describe blocks:
```js
 describe('sanitizeAxes', () => {
  describe('compute numeric extrema', () => {
    // here we go
    it('should do sth', () => { ... })
  });
});
```

File: src/utils/Options.ts (hunk 2 above)
Comment (lorem--ipsum) [inline src/utils/Options.ts]: Right, that's way better. Dunno why I didn't write this. I'll change it right away.

**Comment by chaosmail** [comment]
great contribution, I love how clean everything looks!

**Comment by lorem--ipsum** [comment]
@chaosmail ? :-)

**Comment by chaosmail** [comment]
sorry, I was and I am still very busy :dango: But this one looks great!

**Pull request merged**